rover_wifi/rover_wifi.ino

```diff
@@ -0,0 +1,58 @@
+#include <ESP8266WiFi.h>
+
+const char *ssid =  "Cloudwifi-339-904";
+const char *pass =  "";
+const char homebase = 'H';
+const char notHomebase = 'N';
+bool isConnectedToWifi = false;
+bool atHomebase = false;
+
+WiFiClient client;
+
+void setup() {
+  Serial.begin(9600);
+  while (!Serial) {
+    ; // wait for the serial port to connect. needed for native usb port
+  }
+}
+
+void loop() {
+  // signal that we are at homebase
+  if (Serial.available() && (Serial.read()==homebase)) {
+    atHomebase = true;
+  }
+
+  // we have left homebase
+  if (Serial.available() && (Serial.read()==notHomebase)) {
+    atHomebase = false;
+    isConnectedToWifi = false;
+  }
+
+  if (atHomebase) {
+    if (!isConnectedToWifi) {
+      connect_to_wifi();
+      isConnectedToWifi = true;
+    } else {
+      if (Serial.available()) {
+        // this prints to serial monitor
+        Serial.write(Serial.read());
+        
+        // TODO: send data to database instead
```

Comment:
is this where it will make HTTP request to the API?

Reply:
Yes. I realize that I need to change this into the form of the soil_data struct so that you can send it properly.